src/PFire.Data/Entities/User.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using PFire.Data.Services;
+
+namespace PFire.Data.Entities
+{
+    internal class User : Entity
```

Comment:
Since we're already doing fluent mapping with `OnModelCreating`, I think we should fully commit to it and not use attributes. Then we can utilize these "data" models as our domain models and not need to duplicate them with `xxModel` classes.

Reply:
The one thing I don't like about using the db entities everywhere is that it allows for some problems if you don't query properly, the big one being lazy loading (which I abhor). Using models instead of entities also allows us to consolidate information (like the friend request information + message), so that the consumers need to know less about the database and more about just the info they need.

Reply:
> The one thing I don't like about using the db entities everywhere is that it allows for some problems if you don't query properly

Yeah this can be true if there are a lot of complex interactions with the database. 

I don't like to think of them as DB entities, but rather domain entites, and even store them on a domain project in my solution. The `ChangeTracker` in the background of EF is what houses the real "DB entities" (in the Infrastructure solution). The Infrastructure project is also where we keep EF dependencies and instructs EF how to map from the database to our domain model using fluent mapping.

The `Model` in this structure would be the farthest layer from the database, commonly the HTTP response/request bodies.

XFire definitely does not have many domain rules to be concerned anyway - so no worries👍

Reply:
By the way, if you're really into architecture and patterns. Check out some of the talks with Julie Lerman talking about DDD and entity framework. Really interesting stuff if you're into that sorta thing

Reply:
I think this is the video that convinced me https://www.youtube.com/watch?v=Z62cbp61Bb8

Reply:
I kept the `Models` but moved them (and `PFireDatabase`) up a level - `Infrastructure` still exposes the entities but we explicitly convert them to models to reduce the amount of knowledge the individual messages need to have.